contracts/contracts/ERC677/starkgate_token.cairo

```diff
@@ -0,0 +1,195 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.uint256 import (
+    Uint256,
+    uint256_add,
+    uint256_check,
+    uint256_le,
+    uint256_lt,
+    uint256_sub,
+)
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.bool import FALSE, TRUE
+from openzeppelin.introspection.erc165.library import ERC165
+from contracts.starkware.starknet.std_contracts.ERC20.ERC20_base import (
+    ERC20_allowances,
+    ERC20_approve,
+    ERC20_burn,
+    ERC20_initializer,
+    ERC20_mint,
+    ERC20_transfer,
+    allowance,
+    balanceOf,
+    decimals,
+    name,
+    symbol,
+    totalSupply,
+)
+from contracts.starkware.starknet.std_contracts.ERC20.permitted import (
+    permitted_initializer,
+    permitted_minter,
+    permitted_minter_only,
+    permittedMinter,
+)
+from contracts.starkware.starknet.std_contracts.upgradability_proxy.initializable import (
+    initialized,
+    set_initialized,
+)
+from contracts.ERC677.library import ERC677
+from contracts.ERC677.interfaces.IERC677Receiver import IERC677Receiver
+
+const CONTRACT_IDENTITY = 'ERC20'
+const CONTRACT_VERSION = 1
+
+@view
+func get_version() -> (version : felt):
+    return (version=CONTRACT_VERSION)
+end
+
+@view
+func get_identity() -> (identity : felt):
+    return (identity=CONTRACT_IDENTITY)
+end
```

Comment:
StarkNet contracts implement something similar to our `type_and_version` interface, using two methods:
- https://github.com/starkware-libs/starkgate-contracts/search?q=get_identity
- https://github.com/starkware-libs/starkgate-contracts/search?q=get_version

This interface doesn't look like a requirement, but it's just their convention, and the current usage is only off-chain in tests. 

1. Please double check and remove if unnecessary.
2. Please define the `ITypeAndVersion` (Chainlink convention) interface, and implement it for every contract.